files/tedge/self_update.toml

```
@@ -31,8 +31,23 @@ on_exec = "resume-update"
 [resume-update]
 # Wait before verifying to give the container updater to verify the image
 script = "sh -c 'sudo -E tedge-container self list && sleep 60'"
+on_success = "needs_restart"
+on_error = "needs_restart"
+
+[needs_restart]
+script = """sh -c '[ "${TEDGE_RESTART_ON_SELF_UPDATE:-}" = 1 ] && exit 2'"""
```

There is a mistake (from my previously submitted suggestion).

```suggestion
script = """sh -c '[ "${TEDGE_RESTART_ON_SELF_UPDATE:-}" = 1 ] && exit 2 || exit 0'"""
```